File: HW/HW9/HW9.java
public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
      TreeNode curr = root;
      int pVal = p.val;
      int qVal = q.val;
      while(true){
          if(pVal < curr.val && qVal < curr.val){
              curr = curr.left;
          } else if(pVal > curr.val &&  qVal > curr.val){
              curr = curr.right;
          } else {
              return curr;
          }
      }
}

File: HW/HW9/HW9.py
def lowestCommonAncestor(self, root: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
        curr = root
        pVal = p.val
        qVal = q.val

        while True:
            if pVal < curr.val and qVal < curr.val:
                curr = curr.left
            elif pVal > curr.val and qVal > curr.val:
                curr = curr.right
            else:
                return curr
